### ISSUE TITLE
DefaultComesLast: new option skipIfLastAndSharedWithCase to raise violation if default doesn't share case

### ISSUE THREAD

**Issue by robertpainsi** [checkstyle/checkstyle#4078]
The [DefaultComesLast](http://checkstyle.sourceforge.net/config_coding.html#DefaultComesLast) check should have a new option to allow the `default` case not to be the last one if it shares another case. e.g.

```java
$ javac MyClass.java 
$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
    <module name="TreeWalker">
      <module name="DefaultComesLast"/>
    </module>
</module>

$ cat MyClass.java 
public class MyClass {
    int mode = 4;
    int method2() {
        switch (mode) {
            case 1:
            default: // No violation with the new option is expeced
                return 1;
            case 2:
                return 2;
            case 3:
                return 3;
        }
     }
}

$ java -jar checkstyle-7.6-all.jar -c config.xml MyClass.java 
Starting audit...
[WARN] /var/tmp/MyClass.java:6:13: Default should be last label in the switch. [DefaultComesLast]
Audit done.

```

The `default` statement shares `case 1`. This is useful if the cases are named (e.g enum) or have an order (e.g. natural, alphabetically). However, if the `default` case doesn't share a case implementation, it should be at the last position.

**Comment by sagar-shah94** [comment]
I am on it

**Comment by romani** [comment]
This issue was approved only because there is request for new property to allow user case.

Without new property such update damage whole idea of the check.
I personally do not like default to be somewhere in the middle - it is hard to catch when switch is big and have big blocks for cases.  In checkstyle code this property will never be used.

Author of request might be right in his context (which I do not know), and agree that there might be cases like this where it make sense. So I agree on new property.

my proposal for property name: `skipIfLastAndSharedWithCase`
@robertpainsi , @rnveach  - please propose better name if you have.

use cases:
```
        switch (mode) {
            case 1:
            default: // No violation with the new option is expeced
                return 1;
        }
        switch (mode) {
            default: // violation !!!! with the new option is expected
            case 1:
                return 1;
        }
```

**Comment by romani** [comment]
fix is merged

**Comment by robertpainsi** [comment]
@sagar-shah94, @romani, @rnveach, tested it, works like a charm! Thank you very much! :kissing_heart:

**Comment by sagar-shah94** [comment]
@robertpainsi Glad to know that. 
Thanks .
